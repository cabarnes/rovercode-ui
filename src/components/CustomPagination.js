import React from 'react';
import { Icon, Pagination } from 'semantic-ui-react';
import { hot } from 'react-hot-loader';

const CustomPagination = (props) => (
  <Pagination
    {...props}
    ellipsisItem={{ content: <Icon name="ellipsis horizontal" />, icon: true }}
    firstItem={{ content: <Icon name="angle double left" />, icon: true }}
    lastItem={{ content: <Icon name="angle double right" />, icon: true }}
    prevItem={{ content: <Icon name="angle left" />, icon: true }}
    nextItem={{ content: <Icon name="angle right" />, icon: true }}
  />
);

export default hot(module)(CustomPagination);
