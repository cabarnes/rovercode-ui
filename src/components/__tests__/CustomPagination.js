import React from 'react';
import { shallow } from 'enzyme';
import CustomPagination from '../CustomPagination';

describe('The CustomPagination component', () => {
  test('should render on the page with no errors', () => {
    const wrapper = shallow(
      <CustomPagination
        defaultActivePage={1}
        totalPages={5}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
