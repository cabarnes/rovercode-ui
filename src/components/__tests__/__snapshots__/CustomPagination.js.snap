// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CustomPagination component should render on the page with no errors 1`] = `
<Pagination
  aria-label="Pagination Navigation"
  boundaryRange={1}
  defaultActivePage={1}
  ellipsisItem={
    Object {
      "content": <Icon
        as="i"
        name="ellipsis horizontal"
      />,
      "icon": true,
    }
  }
  firstItem={
    Object {
      "content": <Icon
        as="i"
        name="angle double left"
      />,
      "icon": true,
    }
  }
  lastItem={
    Object {
      "content": <Icon
        as="i"
        name="angle double right"
      />,
      "icon": true,
    }
  }
  nextItem={
    Object {
      "content": <Icon
        as="i"
        name="angle right"
      />,
      "icon": true,
    }
  }
  pageItem={Object {}}
  prevItem={
    Object {
      "content": <Icon
        as="i"
        name="angle left"
      />,
      "icon": true,
    }
  }
  siblingRange={1}
  totalPages={5}
/>
`;
